packages/db/tests/search/test-CollectionSearch.js

```diff
@@ -0,0 +1,216 @@
+
+const {
+  destroyLocalTestDb,
+  generateLocalTestDb,
+  CollectionPgModel,
+  fakeCollectionRecordFactory,
+  migrationDir,
+} = require('../../dist');
+
+const testDbName = `collection_${cryptoRandomString({ length: 10 })}`;
+
+test.before(async (t) => {
+  const { knexAdmin, knex } = await generateLocalTestDb(
+    testDbName,
+    migrationDir
+  );
+
+  t.context.knexAdmin = knexAdmin;
+  t.context.knex = knex;
+
+  t.context.collectionPgModel = new CollectionPgModel();
+  const collections = [];
+  range(100).map((num) => (
+    collections.push(fakeCollectionRecordFactory({
+      name: num % 2 === 0 ? `testCollection___00${num}` : `fakeCollection___00${num}`,
+      version: `${num}`,
+      cumulus_id: num,
+      updated_at: new Date(1579352700000 + (num % 2) * 1000),
+      process: num % 2 === 0 ? 'ingest' : 'publish',
+      report_to_ems: num % 2 === 0,
+    }))
+  ));
+
+  await t.context.collectionPgModel.insert(
+    t.context.knex,
+    collections
+  );
+});
+
+test.after.always(async (t) => {
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
+test('CollectionSearch supports page and limit params', async (t) => {
+  const { knex } = t.context;
+  let queryStringParameters = {
+    limit: 20,
+    page: 2,
+  };
+  let dbSearch = new CollectionSearch({ queryStringParameters });
+  let response = await dbSearch.query(knex);
+  t.is(response.meta.count, 100);
+  t.is(response.results?.length, 20);
+
+  queryStringParameters = {
+    limit: 11,
+    page: 10,
+  };
+  dbSearch = new CollectionSearch({ queryStringParameters });
+  response = await dbSearch.query(knex);
+  t.is(response.meta.count, 100);
+  t.is(response.results?.length, 1);
+
+  queryStringParameters = {
+    limit: 10,
+    page: 11,
+  };
+  dbSearch = new CollectionSearch({ queryStringParameters });
+  response = await dbSearch.query(knex);
+  t.is(response.meta.count, 100);
+  t.is(response.results?.length, 0);
+});
+
+test('CollectionSearch returns correct response for basic query', async (t) => {
+  const { knex } = t.context;
+  const AggregateSearch = new CollectionSearch();
+  const results = await AggregateSearch.query(knex);
+  t.is(results.meta.count, 100);
+  t.is(results.results.length, 10);
+});
+
+test('CollectionSearch supports infix and prefix search', async (t) => {
+  const { knex } = t.context;
+  let queryStringParameters = {
+    limit: 20,
+    infix: 'test',
+  };
+  const dbSearch = new CollectionSearch({ queryStringParameters });
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 50);
+  t.is(response.results?.length, 20);
+
+  queryStringParameters = {
+    limit: 20,
+    prefix: 'fake',
+  };
+  const dbSearch2 = new CollectionSearch({ queryStringParameters });
+  const response2 = await dbSearch2.query(knex);
+  t.is(response2.meta.count, 50);
+  t.is(response2.results?.length, 20);
+});
+
+test('CollectionSearch supports term search', async (t) => {
```

Comment:
Since you have a `term search for date field` case below, and since granules has it broken out, maybe we can be more specific about the search terms in the description, and have each of them separate?